src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static partial class OperationExtensions
+    {
+        /// <summary>
+        /// Returns the <see cref="ValueUsageInfo"/> for the given operation.
+        /// This extension can be removed once https://github.com/dotnet/roslyn/issues/25057 is implemented.
+        /// </summary>
+        public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
+        {
+            /*
+            |    code         | Read | Write | ReadableRef | WritableRef |
+            | nameof(x)       |      |       |             |             |
+            | x.Prop = 1      |      |  ✔️   |             |             |
+            | x.Prop += 1     |  ✔️  |  ✔️   |             |             |
+            | x.Prop++        |  ✔️  |  ✔️   |             |             |
+            | Foo(x.Prop)     |  ✔️  |       |             |             |
+            | Foo(x.Prop)*    |      |       |     ✔️      |             |
+            | Foo(out x.Prop) |      |       |             |     ✔️      |
+            | Foo(ref x.Prop) |      |       |     ✔️      |     ✔️      |
+
+            * where void Foo(in T v)
+            */
+
+            if (operation.Parent is IAssignmentOperation assignmentOperation &&
+                assignmentOperation.Target == operation)
```

Comment:
i'm very concerned here about both implicit conversions *and* parenthesized expressions.  what if i have: 

```
(x.y) = 4
```

?

Reply:
Both the cases work with the current implementation due to the shape of the operation tree. I will add unit tests.

---
In reply to: [212736397](https://github.com/dotnet/roslyn/pull/29511#discussion_r212736397) [](ancestors = 212736397)

Reply:
That's actually interesting to me.  I thought the IOp tree contained parenthesized ops and conversio ops...  does 'target' bypass that?

Reply:
Conversion is actually wrapping the assigned value. Parenthesized operation is never generated for C# and only present for VB (as parenthesis can alter semantics in VB), but VB syntax does not allow parenthesis on the assignment target.

---
In reply to: [212782545](https://github.com/dotnet/roslyn/pull/29511#discussion_r212782545) [](ancestors = 212782545)